[18.0][MIG] purchase_stock_operating_unit: Migration to 18.

BhaveshHeliconia opened the pull request.

Saran440 committed:
[15.0][ADD] purchase_stock_operating_unit

Commit:
[UPD] Update purchase_stock_operating_unit.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

BhaveshHeliconia committed:
[IMP] purchase_stock_operating_unit: pre-commit auto fixes

BhaveshHeliconia committed:
[MIG] purchase_stock_operating_unit: Migration to 18.0

kanisorn-kate reviewed (approved):
LGTM

Saran440 commented on purchase_stock_operating_unit/models/purchase.py:
```diff
@@ -1,24 +1,63 @@
 class PurchaseOrder(models.Model):
     _inherit = "purchase.order"
 
+    picking_type_id = fields.Many2one(
```
Functional Test :+1: 

But I'm not sure. Should we use onchange instead compute with `picking_type_id` and `operating_unit_id`?

AaronHForgeFlow replied:
I think onchange makes more sense. Maybe a constraint if there is inconsistency with the operating unit.

BhaveshHeliconia commented:
I have addressed your feedback @Saran440 and @AaronHForgeFlow. Could you please review?

1. Refactored `picking_type_id` and `operating_unit_id` from computed fields to regular fields with `onchange` methods.
2. Kept the Python constraint to enforce consistency between the Operating Unit and the Picking Type.
3. Updated the test suite to use `Form` objects so the onchange logic is properly triggered and verified during testing.

Saran440 reviewed (approved):
Code Review 👍

AaronHForgeFlow reviewed (approved):
Code review LGTM

AaronHForgeFlow commented:
/ocabot merge nobump
